sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -112,8 +131,31 @@ public HttpPipeline getHttpPipeline() {
             });
     }
 
-    // TODO: This is a temporary implementation for sample purposes. This should be spruced up/replaced once this API is actually designed.
-    // Input is Object and output is Response<T>.
+    /**
+     * Creates a digital twin.
+     *
+     * @param digitalTwinId The Id of the digital twin.
+     * @param digitalTwin The application/json digital twin to create.
+     * @param clazz The model class to convert the response to.
+     * @param <T> The generic type to convert the response to.
+     * @return The application/json digital twin created.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public <T> Mono<T> createDigitalTwin(String digitalTwinId, Object digitalTwin, Class<T> clazz)
+    {
+        return createDigitalTwinWithResponse(digitalTwinId, digitalTwin, clazz)
+            .map(DigitalTwinsResponse::getValue);
+    }
+
+    /**
+     * Creates a digital twin.
+     *
+     * @param digitalTwinId The Id of the digital twin.
+     * @param digitalTwin The application/json digital twin to create.
+     * @param clazz The model class to convert the response to.
+     * @param <T> The generic type to convert the response to.
+     * @return A Http response containing application/json digital twin created.
+     */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public <T> Mono<DigitalTwinsResponse<T>> createDigitalTwinWithResponse(String digitalTwinId, Object digitalTwin, Class<T> clazz) {
```

Comment:
Shouldn't ```Object digitalTwin``` be ```T digitalTwin``` in the parameter list?

Reply:
The input digital twin object isn't being serialized or deserialized to any type, to the `<T>` type information doesn't lend any value. So we can take it in as an `Object` directly, just that it needs to be json serializable.

Reply:
I'd say that taking in T rather than Object forces the user to understand that they will be receiving the same type that they are providing though, right?

Reply:
It would be a confusing error to the user if they provided a non-T object as an input parameter here